narwhals/stable/v1.py

```diff
@@ -528,26 +528,26 @@ def from_native(
 
 @overload
 def from_native(
-    native_dataframe: IntoDataFrameT | T,
+    native_dataframe: Any,
     *,
     strict: Literal[False],
     eager_only: None = ...,
     eager_or_interchange_only: Literal[True],
     series_only: None = ...,
     allow_series: None = ...,
-) -> DataFrame[IntoDataFrameT] | T: ...
+) -> Any: ...
```

Comment:
drive-by, but I can't get these annotations to work reliably for `strict=False`

I think that might be OK though, because in those cases, people are going to have to do `isinstance` checks anyway?

We can always tighten later, but for now, this would fail the following:
```
from narwhals.typing import IntoDataFrame

def bar(df: dict | IntoDataFrame) -> None:
    df_nw = nw.from_native(df, strict=False, eager_only=True)
```

```
t.py:35: error: No overload variant of "from_native" matches argument types "dict[Any, Any] | NativeFrame | DataFrame[Any] | DataFrameLike", "bool", "bool"  [call-overload]
t.py:35: note: Possible overload variants:
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[False], eager_only: None = ..., eager_or_interchange_only: Literal[True], series_only: None = ..., allow_series: Literal[True]) -> Any
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[False], eager_only: Literal[True], eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: Literal[True]) -> Any
t.py:35: note:     def [IntoDataFrameT, T] from_native(native_dataframe: IntoDataFrameT | T, *, strict: Literal[False], eager_only: None = ..., eager_or_interchange_only: Literal[True], series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoDataFrameT] | T
t.py:35: note:     def [IntoDataFrameT, T] from_native(native_dataframe: IntoDataFrameT | T, *, strict: Literal[False], eager_only: Literal[True], eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoDataFrameT] | T
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[False], eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: Literal[True]) -> Any
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[False], eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: Literal[True], allow_series: None = ...) -> Any
t.py:35: note:     def [IntoFrameT, T] from_native(native_dataframe: IntoFrameT | T, *, strict: Literal[False], eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoFrameT] | LazyFrame[IntoFrameT] | T
t.py:35: note:     def [IntoDataFrameT] from_native(native_dataframe: IntoDataFrameT, *, strict: Literal[True] = ..., eager_only: None = ..., eager_or_interchange_only: Literal[True], series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoDataFrameT]
t.py:35: note:     def [IntoDataFrameT] from_native(native_dataframe: IntoDataFrameT, *, strict: Literal[True] = ..., eager_only: Literal[True], eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoDataFrameT]
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[True] = ..., eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: Literal[True]) -> DataFrame[Any] | LazyFrame[Any] | Series
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: Literal[True] = ..., eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: Literal[True], allow_series: None = ...) -> Series
t.py:35: note:     def [IntoFrameT] from_native(native_dataframe: IntoFrameT, *, strict: Literal[True] = ..., eager_only: None = ..., eager_or_interchange_only: None = ..., series_only: None = ..., allow_series: None = ...) -> DataFrame[IntoFrameT] | LazyFrame[IntoFrameT]
t.py:35: note:     def from_native(native_dataframe: Any, *, strict: bool, eager_only: bool | None, eager_or_interchange_only: bool | None = ..., series_only: bool | None, allow_series: bool | None) -> Any
```

Reply:
@FBruzzesi any ideas here, or just ignore this (`strict=False`) case for now?

Reply:
Taking a quick look

Reply:
thanks - i've noticed that actually it doesn't strictly need to be done now, we can leave it til later